api-metastore/src/main/java/org/zalando/nakadi/service/AvroSchemaCompatibility.java

```diff
@@ -0,0 +1,153 @@
+package org.zalando.nakadi.service;
+
+import org.apache.avro.AvroRuntimeException;
+import org.apache.avro.Schema;
+import org.apache.avro.SchemaCompatibility;
+import org.apache.avro.SchemaCompatibility.Incompatibility;
+import org.apache.avro.SchemaCompatibility.SchemaPairCompatibility;
+import org.springframework.stereotype.Component;
+import org.zalando.nakadi.domain.CompatibilityMode;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component
+public class AvroSchemaCompatibility {
```

Comment:
general comment, it is hard to read the functional style here (maybe it is just me), but it gets to verbose in Java

Reply:
for transitive it would be a bit different, but in nutshell what is done:
```
SchemaCompatibility.checkReaderWriterCompatibility(newSchema, getLastSchema.get())
.getResult().getIncompatibilities().
.map(I -> new AvroIncompatibility(incompat.getLocation(), incompat.getMessage(), incompat.getType()))
.collect(Collectors.toList());
```

Reply:
yes, correct.